src/convertToObject.js

```diff
@@ -6,7 +6,30 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const cleanedString = sourceString
+    .replace(/^\s*[\r\n]/gm, '')
+    .replace(/\s*;\s*/g, ';')
+    .replace(/\s*:\s*/g, ': ')
+    .trim();
+
+  const keyValuePairs = cleanedString
+    .split(';')
+    .filter((elem) => elem.length > 0);
+
+  return keyValuePairs.reduce((acc, val) => {
+    const properties = val.split(':');
+
+    if (properties.length !== 2) {
+      return acc;
+    }
+
+    const key = properties[0].trim();
+    const value = properties[1].trim().replace(';', '');
```

Comment:
The `.replace(';', '')` is unnecessary here since the semicolon is already used as a delimiter in the `split` method. Removing it from the value might lead to incorrect data if semicolons are part of the value.